packages/@atjson/editor/src/index.ts

```diff
@@ -1,184 +1,168 @@
-import { HIR } from '@atjson/hir';
 import Document from '@atjson/document';
+import WebComponentRenderer from './webcomponent-renderer';
 import events from './mixins/events';
 import './text-selection';
+import './text-input';
+import './selection-toolbar';
 
 const TEXT_NODE_TYPE = 3;
+type Range = { start: number, end: number };
 
 type Element = TextNode | HTMLElement;
 
-function getNodeAndOffset(node: Element | null, offset: number): { node: Element | null, offset: number } | never {
-  if (node == null) {
-    return { node: null, offset };
-  } else if (node.nodeType === TEXT_NODE_TYPE) {
-    return { node, offset };
-  } else if (node.childNodes.length > 0) {
-    let offsetNode = node.childNodes[offset];
-
-    // If the offset node has a single child node,
-    // use that node instead of the parent
-    if (offsetNode.nodeType !== TEXT_NODE_TYPE &&
-        offsetNode.childNodes.length === 1) {
-      offsetNode = offsetNode.childNodes[0];
-    }
+export default class Editor extends events(HTMLElement) {
 
-    // The offset node is a text node; quickly return
-    if (offsetNode.nodeType === TEXT_NODE_TYPE) {
-      return { node: offsetNode, offset: 0 };
+  selection: Range;
 
-    // Find the closest text node and return that
-    } else if (!offsetNode.hasChildNodes()) {
-      let adjustedOffset = offset - 1;
+  static template = '<text-input><text-selection><selection-toolbar slot="toolbar"></selection-toolbar><div class="editor" style="white-space: pre-wrap; padding: 1em; border: 1px solid black; border-radius: 5px; outline: none; font-size: 1.5em;" contenteditable></div></text-selection></text-input>';
```

Comment:
🙅 inline styles 🙅

Reply:
Yeah - This component doesn't inherit from our base component class, so this was easier for the immediate moment. Need to refactor.